nxdrive/gui/folders_dialog.py

```diff
@@ -275,6 +276,17 @@ def __init__(
         self.tree_view.setContextMenuPolicy(Qt.CustomContextMenu)
         self.tree_view.customContextMenuRequested.connect(self.open_menu)
 
+    def keyPressEvent(self, event: QKeyEvent) -> None:
+        """Direct Transfer Handle Esc key on Max state(NXDRIVE-2737).
+        On Esc keypress event, restoring the maximized window to normal state
```

Comment:
Maybe a little more like this: https://github.com/nuxeo/nuxeo-drive/blob/master/nxdrive/engine/tracker.py#L208